dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+            for (Identifier ident : idents) {
+                if (ident.getAssetType().equals("folder")) {
+                    folderIdents.add(ident);
+                }else {
+                    otherIdents.add(ident);
+                }
+            }
+        }
+
+        // sorting folder identifiers by parent path in order to pass parent check
+        Collections.sort(folderIdents, new Comparator<Identifier>() {
+            public int compare(Identifier o1, Identifier o2) {
+                return o1.getParentPath().compareTo(o2.getParentPath());
+            }
+        });
+
+        // saving folder identifiers
+        for (Identifier ident : folderIdents) {
+            if (!SYSTEM_FOLDER_PATH.equals(ident.getParentPath())) {
+                ident.setParentPath(ident.getParentPath());
+                ident.setAssetName(ident.getAssetName());
+            }
+            Logger.info(this, "Importing folder path " + ident.getParentPath() + ident.getAssetName());
+            APILocator.getIdentifierAPI().save(ident);
+        }
+
+        // now we need to save all remaining identifiers (folders already added)
+        for (final Identifier ident : otherIdents) {
+            APILocator.getIdentifierAPI().save(ident);
+        }
+
+
+
+        // we store here defaultFileType for every folder
+        // because of mutual folder <--> structure dependency
+        final Map<String, String> fileTypesInodes = new HashMap<String, String>();
+
+        // now we can import folders
+        for (File ff : contains(Folder.class.getCanonicalName())) {
+            try {
+                doXMLFileImport(ff, new ObjectFilter() {
+                    public boolean includeIt(Object obj) {
+                        Folder f = (Folder) obj;
+                        fileTypesInodes.put(f.getInode(), f.getDefaultFileType());
+                        f.setDefaultFileType(null);
+                        return true;
+                    }
+                });
+            } catch (Exception e) {
+                Logger.error(this, "Unable to load " + ff.getName() + " : " + e.getMessage(), e);
+            }
+        }
+
+
+        // we need content types before contentlets
+        // but content types have references to folders and hosts identifiers
+        // so, here is the place to do it
+        for (File file : endsWith(ContentTypeImportExportUtil.CONTENT_TYPE_FILE_EXTENSION)) {
+            new ContentTypeImportExportUtil().importContentTypes(file);
+        }
+
+        // updating file_type on folder now that structures were added
+        DotConnect dc = new DotConnect();
+        for (Entry<String, String> entry : fileTypesInodes.entrySet()) {
+            dc.setSQL("update folder set default_file_type=? where inode=?");
+            dc.addParam(entry.getValue());
+            dc.addParam(entry.getKey());
+            dc.loadResult();
+        }
+
+
+        for (File file : contains(Relationship.class.getCanonicalName() + "_")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'file' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)